connectors/sources/redis.py

```diff
@@ -0,0 +1,269 @@
+        except Exception as exception:
+            self._logger.warning(f"Database {db} not found. Error: {exception}")
+
+    async def get_databases(self):
+        """Returns number of databases from config_get response
+
+        Returns:
+            list: List of databases
+        """
+        if self.database != ["*"]:
+            for database in set(self.database):
+                yield database
+        else:
+            try:
+                databases = await self.redis_client.config_get("databases")
+                for database in list(range(int(databases.get("databases", 1)))):
+                    yield database
+            except Exception as exception:
+                self._logger.warning(
+                    f"Something went wrong while fetching database list. Error: {exception}",
+                    exc_info=True,
+                )
+
+    async def get_paginated_key(self, pattern, _type=None):
+        """Make a paginated call for fetching database keys.
+
+        Args:
+            pattern: keyname or pattern
+            _type: Datatype for filter data
+
+        Yields:
+            string: key in database
+        """
+        client = await self.redis_client  # pyright: ignore
+        async for key in client.scan_iter(match=pattern, count=PAGE_SIZE, _type=_type):
+            yield key
+
+    async def get_key_value(self, key, key_type):
+        """Fetch value of key for database.
+
+        Args:
+            key: Name of key
+            key_type: Type of key
+
+        Yields:
+            Value of key
+        """
+        try:
+            if key_type == KeyType.HASH.value:
+                return await self.redis_client.hgetall(key)
+            elif key_type == KeyType.STREAM.value:
+                return await self.redis_client.xread({key: "0"})
+            elif key_type == KeyType.LIST.value:
+                return await self.redis_client.lrange(key, 0, -1)
+            elif key_type == KeyType.SET.value:
+                return await self.redis_client.smembers(key)
+            elif key_type == KeyType.ZSET.value:
+                return await self.redis_client.zrange(key, 0, -1, withscores=True)
+            elif key_type == KeyType.JSON.value:
+                value = await self.redis_client.execute_command("JSON.GET", key)
+                return json.loads(value)
+            else:
+                value = await self.redis_client.get(key)
+                if value is not None:
+                    try:
+                        value_json = json.loads(value)
+                        return value_json
+                    except json.JSONDecodeError:
+                        return value
+        except UnicodeDecodeError as exception:
+            self._logger.warning(
+                f"Something went wrong while fetching value for key {key}. Error: {exception}",
+                exc_info=True,
+            )
+            return ""
+
+    async def ping(self):
+        await self.client()
```

Comment:
Every call to `ping` will create a new client

Reply:
I guess ping calls once in a whole running connector and if that is the case then client will also be called once only.